doc/mgr/orchestrator_modules.rst

```diff
@@ -150,6 +150,48 @@ specify a location when creating a stateless service.
 OSD services generally require a specific placement choice, as this
 will determine which storage devices are used.
 
+Error Handling
+--------------
+
+The main goal of error handling within orchestrator modules is to provide debug information to
+assist users when dealing with deployment errors.
+
+In detail, orchestrators need to explicitly deal with different kind of errors:
+
+1. No orchestrator configured
+
+.. autoclass:: NoOrchestrator
+
+2. An orchestrator doesn't implement a specific method.
+
+   For example, an Orchestrator doesn't support ``add_host``.
+
+   In this case, a ``NotImplementedError`` is raised.
+
+3. Missing features within implemented methods.
+
+   e.g. deploying iSCSI gateways is not supported
+
+.. autoclass:: MissingOrchestratorSupport
+
+4. Input validation errors
+
+   the ``orchestrator_cli`` module and other calling modules are supposed to
+   provide meaningful error messages.
+
+5. Errors when actually executing commands
+
+   The resulting Completion should contain an error string that assists in understanding the
+   problem.
```

Comment:
Something like `is_errored() == True` plus a message in `result`?

Reply:
That's effectively what's done today and it works ok.